sympy/solvers/solvers.py

```diff
@@ -361,6 +361,38 @@ def checksol(f, symbol, sol=None, **flags):
     # TODO: improve solution testing
 
 
+def failing_assumptions(expr, **assumptions):
+    """Returns a dictionary of assumptions that does not satisfies `expr`,
```

Comment:
Return a dictionary containing assumptions with values not matching those of the passed assumptions.